pkg/kubelet/gpu/nvidia/nvidia_gpu_manager_test.go

```diff
@@ -36,6 +39,52 @@ func (tapl *testActivePodsLister) GetActivePods() []*v1.Pod {
 	return tapl.activePods
 }
 
+func makeValidTestProcFile() string {
+	content := []byte("aaa\t10debbb\tccc\nddd\teee\tfff\nggg\t10dehhh\tiii\n")
+	tmpfile, err := ioutil.TempFile("", "tmp")
+	if err != nil {
+		fmt.Print(err)
+	}
+	if _, err := tmpfile.Write(content); err != nil {
+		fmt.Print(err)
+	}
+	if err := tmpfile.Close(); err != nil {
+		fmt.Print(err)
+	}
+	devicesFile := tmpfile.Name()
+	return devicesFile
+}
+
+func makeDummyDevDir() string {
+	content := []byte("dummy content")
+	dir, err := ioutil.TempDir("", "tmpdir")
+	if err != nil {
+		fmt.Print(err)
+	}
+	fakeDevices := []string{"nvidia1", "nvidia2"}
+	for _, fileName := range fakeDevices {
+		tmpfn := filepath.Join(dir, fileName)
+		if err := ioutil.WriteFile(tmpfn, content, 0666); err != nil {
+			fmt.Print(err)
+		}
+	}
+	return dir
+}
+
+func TestDiscoverGPUS(t *testing.T) {
```

Comment:
Can you add some more test cases?
- To test behavior when no expected devices.
- To test behavior when there are expected devices but no found devices.